src/preprocessing/passes/ackermann.cpp

```diff
@@ -187,9 +188,124 @@ void collectFunctionsAndLemmas(FunctionToArgsMap& fun_to_args,
 
 /* -------------------------------------------------------------------------- */
 
+bool needsReplace(TNode term)
+{
+  if (term.getType().isSort() && !(term.getKind() == kind::BOUND_VARIABLE))
```

Comment:
I think that we would be better off to just not allow this preprocessing pass to run when quantifiers are enabled (because we remove UF from the logic but we may not eliminate all UFs if we have bound variables). Also, I am not sure if we'd need to do more work to make the pass work with quantifiers in general. What do you think @yoni206 ?

Reply:
Ackermannization assumes no quantifiers, so yes I support not allowing the pass in the presence of quantifiers.